src/widget.rs

```diff
@@ -108,6 +119,48 @@ impl<O: IsA<Widget>> WidgetExtManual for O {
         }
     }
 
+    fn add_tick_callback<P: Fn(&Widget, &gdk::FrameClock) -> bool + 'static>(
+        &self,
+        callback: P,
+    ) -> TickCallbackId {
+        // Generated by gir, modified for wrapping in TickCallbackId
+        let callback_data: Box<P> = Box::new(callback);
+        unsafe extern "C" fn callback_func<P: Fn(&Widget, &gdk::FrameClock) -> bool + 'static>(
+            widget: *mut ffi::GtkWidget,
+            frame_clock: *mut gdk_ffi::GdkFrameClock,
+            user_data: glib_ffi::gpointer,
+        ) -> glib_ffi::gboolean {
+            let widget = from_glib_borrow(widget);
+            let frame_clock = from_glib_borrow(frame_clock);
+            let callback: &P = &*(user_data as *mut _);
+            let res = (*callback)(&widget, &frame_clock);
+            res.to_glib()
+        }
+        let callback = Some(callback_func::<P> as _);
+        unsafe extern "C" fn notify_func<P: Fn(&Widget, &gdk::FrameClock) -> bool + 'static>(
+            data: glib_ffi::gpointer,
+        ) {
+            let _callback: Box<P> = Box::from_raw(data as *mut _);
+        }
+        let destroy_call3 = Some(notify_func::<P> as _);
+        let super_callback0: Box<P> = callback_data;
+        TickCallbackId(unsafe {
+            ffi::gtk_widget_add_tick_callback(
+                self.as_ref().to_glib_none().0,
+                callback,
+                Box::into_raw(super_callback0) as *mut _,
+                destroy_call3,
+            )
+        })
+    }
+
+    fn remove_tick_callback(&self, id: TickCallbackId) {
```

Comment:
Not better add this function to `TickCallbackId` ?

Reply:
And please add `#[allow(clippy::needless_pass_by_value)]` if clippy nags.

Reply:
@GuillaumeGomez, @sdroege You agree to moving remove function to type itself?

Reply:
Works for me

Reply:
Done, but `TickCallbackId` no longer implements Debug, Eq, PartialEq, and Send because WeakRef<Widget> doesn't.